packages/router/useRouteParams/index.ts

```diff
@@ -64,7 +64,7 @@ export function useRouteParams<
         if (_params.get(name) === v)
           return
 
-        _params.set(name, v)
+        _params.set(name, (v === defaultValue || v === null) ? undefined : v)
```

Comment:
```suggestion
        _params.set(name, v == null ? undefined : v)
```

Reply:
But this breaks it again in case it is the `defaultValue` or did you just meant to change the last part `=== null` and leave the first part as is?

```suggestion
        _params.set(name, (v === defaultValue || v == null) ? undefined : v)
```

Reply:
At least my added test cases break